pkg/exec/devfile.go

```diff
@@ -0,0 +1,71 @@
+package exec
+
+import (
+	"fmt"
+
+	adaptersCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
+	"github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/log"
+)
+
+// ExecuteDevfileBuildAction executes the devfile build command action
+func ExecuteDevfileBuildAction(client ExecClient, action common.DevfileCommandAction, commandName string, compInfo adaptersCommon.ComponentInfo, show bool) error {
+	var s *log.Status
+
+	// Change to the workdir and execute the command
+	var cmdArr []string
+	if action.Workdir != nil {
+		cmdArr = []string{"/bin/sh", "-c", "cd " + *action.Workdir + " && " + *action.Command}
```

Comment:
i feel like `/bin/sh` should be a const ?

Reply:
also why are you combining the commands? you shouldn't use busing `+`. It should be like so:

```go
[]string{"/bin/sh", "-c", "cd ", *action.Workdir, "&&" , *action.Command}
```

etc

Reply:
I did that way initially but I think there was a reason i did this when i wrote it first for kube, let me try it out again..

Reply:
the reason why it is this way - `"cd /tmp && pwd"`, rather than `"cd", "/tmp", "&&", "pwd"` is because we are using `/bin/sh -c` to execute the command, as a result they need to be in the within a single `""`. I have put up a comment before it to explain why we're doing this.

I have tried to run exec without `/bin/sh` but it seems the kube client api is error'ring out. Since both docker and kube works this way, I have left at this. We can clean this up in the future, if we find a better way. Would you like an issue to track it? @cdrage

Reply:
tracking with https://github.com/openshift/odo/issues/3015 if this can be improved